### DISCUSSION TITLE
Unused Spree::Order#temporary_payment_source attribute

### DISCUSSION THREAD

**Discussion by AlessioRocco** [solidusio/solidus#4554] [Troubleshooting]
The `Spree::Order` attribute `temporary_payment_source` seems not to be set in any place.
The only line where it's used is:

https://github.com/solidusio/solidus/blob/475d9db5d0291dd4aeddc58ec919988c336729bb/core/app/models/spree/wallet/add_payment_sources_to_wallet.rb#L16

Do you have any idea on what the purpose of that attribute?

**Comment by peterberkenbosch** [comment]
it was used to have a choice to have the payment source saved or not. It has since then been unused indeed. Current wip around this attribute is being done here: https://github.com/solidusio/solidus/pull/2965

**Comment by kennyadsl** [comment]
I think that saving the source should be done per payment and not per order.

There's no guarantee that the user enters a single payment/source per order, there could be frontends that allow adding more than one credit card in the payment step and user could want to save just a subset of them.

I think we should move the `temporary` attribute on payments or sources calling it with a better name as well (like `single_usage_allowed`), this way it will be more flexible.

WDYT?

**Comment by peterberkenbosch** [comment]
Good call @kennyadsl, I totally agree! Thanks. Will update this PR asap
